antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/Lexer.kt

```diff
@@ -18,14 +18,16 @@ import org.antlr.v4.kotlinruntime.misc.Interval
 abstract class Lexer : Recognizer<Int, LexerATNSimulator>, TokenSource {
 
 
-//    /** Set the char stream and reset the lexer  */
-//    fun setInputStream(input: IntStream) {
-//        this.inputStream = null
-//        this._tokenFactorySourcePair = Pair<TokenSource, CharStream>(this, inputStream)
-//        reset()
-//        this.inputStream = input as CharStream
-//        this._tokenFactorySourcePair = Pair<TokenSource, CharStream>(this, readInputStream())
-//    }
+    /**
+     * Set the char stream and reset the lexer.
+     */
+    public fun setInputStream(input: IntStream) {
+        this.inputStream = null
```

Comment:
@ftomassetti do you recall why you moved the `inputStream` field to the `Recognizer` class?  
On the Java runtime Lexer has its own `_input`, and `Parser` only has the `TokenStream`.

If you look into `DefaultErrorStrategy.recover` here, you'll see a line:
```
if (lastErrorIndex == recognizer.readInputStream()!!.index() ...
```
where `recognizer.readInputStream()` returns the `Recognizer.inputStream` (`IntStream`).

If you look into the Java runtime, you'll see `recognizer.getInputStream`, which returns `TokenStream`.  
I feel like this is a bug.

Reply:
Solved with 7b761e9d4cc9a229b1b03d6af269b68eed996fa2  
Have a look at that commit and let me know if it feels ok or not. To me we also get better type correctness compared to Java now.

Reply:
> @ftomassetti do you recall why you moved the inputStream field to the Recognizer class?

No, I am afraid it was a few years ago and I forgot. It may well be a bug